util/httpclient/utils.go

```diff
@@ -42,6 +46,21 @@ func init() {
 			MinVersion: tls.VersionTLS12,
 		},
 	}
+	// If HTTP2_CLIENT_PING_ENABLED env var is set, the transport will setup http2 ping
+	v, _ := strconv.ParseBool(os.Getenv("HTTP2_CLIENT_PING_ENABLED"))
+	if v {
+		tr2 := defaultTransport.Clone()
+		if http2Transport, err := http2.ConfigureTransports(tr2); err == nil {
+			// if the connection has been idle for 3 seconds, send a ping frame for a health check
```

Comment:
would 3s/2s be too short? How about just following the azure go sdk settings using 10s / 5s? 

https://github.com/Azure/azure-sdk-for-go/blob/b9db6714434b3704b22f1d5d7453577c94508c25/sdk/azcore/runtime/transport_default_http_client.go#L40-L43

Reply:
Adjusting to the http client setup in guard (3sec timeout)
https://github.com/kubeguard/guard/blob/04de5608918fde6cf0dd4824c81221e73c5fabd8/util/azure/utils.go#L514

I'm not sure why go ask set these values, in our RPs we do setup around these time as well and we are not seeing any issue. Most of the time, if a connection is stale for more than 3sec, no harm to close it and re open a new one no ?